### PR TITLE
feat: Add top-level-interop rule

### PR DESCRIPTION
**Pull request by hildjj**
A new rule that checks that the top-level item in a JSON file is either an array or an object, as suggested by RFC8259.  Not turned on in the recommended config because the software that had this issue is almost entirely obsolete.

Refs #68.

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

As discussed in #68, add a new rule to enforce the top-level item in a JSON doc be an array or object.

#### What changes did you make? (Give an overview)

Added src/rules/top-level-interop.js and tests/rules/top-level-interop.test.js

#### Related Issues

Refs #68

#### Is there anything you'd like reviewers to focus on?

Should this be in the recommended set?  I could be persuaded either way.

### CODE REVIEW

**Comment by hildjj** [comment]
I clicked through the CLA.

File: src/index.js
Comment (nzakas) [inline src/index.js:16]:
```suggestion
import topLevelInterop from "./rules/top-level-interop.js";
```

File: src/index.js
Comment (nzakas) [inline src/index.js:37]:
```suggestion
		"top-level-interop": topLevelInterop,
```

File: src/rules/top-level-interop.js
Comment (nzakas) [inline src/rules/top-level-interop.js:12]: We generally use "require" unless the rule begins with `no-`.

```suggestion
				"Require the top-level value to be an array or object",
```

File: src/rules/top-level-interop.js
Comment (nzakas) [inline src/rules/top-level-interop.js:17]: We tend to use single quotes to surround values in message.

```suggestion
				"Top level item should be array or object, got '{{type}}'.",
```

**Review by nzakas — CHANGES_REQUESTED** [review]
Thanks for putting this together. Overall, it looks good. Just some areas to clean up before merging.

**Review by nzakas — DISMISSED** [review]
LGTM. Waiting for a second review before merging.

File: src/index.js
Comment (mdjermanovic) [inline src/index.js:47]:
```suggestion
```

I think we should just omit it from the recommended config, as disabling it sounds like "it's recommended not to use this rule". @nzakas what do you think?

File: src/rules/top-level-interop.js
Comment (mdjermanovic) [inline src/rules/top-level-interop.js:17]:
```suggestion
				"Top level item should be array or object, got '{{type}}'.",
```

Per https://github.com/eslint/json/pull/69#discussion_r1868308503. This will also fix the formatting error.

**Review by nzakas — APPROVED** [review]
LGTM. Waiting for @mdjermanovic to re-review.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
